### PR TITLE
Create VSourcesTable component

### LINKED ISSUE #1088: Create table component; convert styles to tailwind
## Problem

Currently, the table on the [sources page](https://wordpress.org/openverse/sources) relies on custom Sass styles. Additionally, all the logic for this table is included directly in the page. As part of our [milestone to remove legacy styles](https://github.com/WordPress/openverse-frontend/milestone/13), we should convert this component to use Tailwind for styles: inline in the templates where possible, but using `@apply {}` declarations where needed. 

We should also make a simple `VTable` component that extracts the current table and styles into a standalone component. From personal experience, table data structures and prop API design can be very controversial 😆 . Whoever implements this should discuss their proposed design prior to building.

**Edit:** Alternatively, this table could, at first, be hardcoded with only the necessary data for this page, and be called the `VSourcesTable`. We could then wait until tables appear again in our designs to write a full table abstraction. I think this is actually a better approach.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1088 by @zackkrida
Related to #1155 by @obulat (not fixes because no design decisions have been made yet)

## Description

This PR creates a `VSourcesTable` that extracts all the table styles. It uses the second approach from the linked issue: instead of trying to implement a very generic table, it only creates a component specific for the Sources table. If/when we have other tables, we can either create a new component, or extract common styles/behaviors from this component.

Other changes include:
- TS-ification
- Removal of `table.scss`
- Addition of Audio Sources table
- Style changes to match the mockups: table row background set `dark-charcoal-06`, font size set to 14px

## Testing Instructions

Open the Sources page, try changing the sorting in the table.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VSourcesTable.vue [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        <td class="text-right">
+          {{ getLocaleFormattedNumber(provider.media_count || 0) }}
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script lang="ts">
+import sortBy from 'lodash.sortby'
```
Comment (sarayourfriend) [inline src/components/VSourcesTable.vue:60]: TIL `lodash.sortby` is over 2300 LOC :scream: Good thing we have #1108 :grin:

File: src/styles/vocabulary.scss [hunk 2]
```diff
@@ -19,5 +19,4 @@
 @import 'components/image';
 @import 'components/navbar';
 @import 'components/card';
-@import 'components/table';
```
Comment (sarayourfriend) [inline src/styles/vocabulary.scss:22]: Yesssssss :tada:

File: src/pages/sources.vue [hunk 3]
```diff
@@ -74,178 +74,35 @@
...
-          <td class="number-cell">
-            {{ getLocaleFormattedNumber(imageProvider.media_count || 0) }}
-          </td>
-        </tr>
-      </tbody>
-    </table>
+    <h5>{{ $t('sources.heading.image') }}</h5>
+    <VSourcesTable media="image" />
+    <h5>{{ $t('sources.heading.audio') }}</h5>
+    <VSourcesTable media="audio" />
```
Comment (sarayourfriend) [inline src/pages/sources.vue:80]: Nice!! :star_struck:

**Review by sarayourfriend — APPROVED** [review]
Excellent!

File: src/components/VSourcesTable.vue [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        >
+          <span class="w-full flex flex-row items-center justify-between">
+            {{ $t('sources.providers.domain') }}
+            <TableSortIcon :active="sorting.field === 'source_url'" />
+          </span>
+        </th>
+        <th
+          tabindex="0"
+          @click="sortTable('media_count')"
+          @keypress.enter="sortTable('media_count')"
```
Comment (obulat) [inline src/components/VSourcesTable.vue:32]:
```suggestion
          @click="sortTable('media_count')"
```

File: src/components/VSourcesTable.vue [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        >
+          <span class="w-full flex flex-row items-center justify-between">
+            {{ $t('sources.providers.source') }}
+            <TableSortIcon :active="sorting.field === 'display_name'" />
+          </span>
+        </th>
+        <th
+          tabindex="0"
+          @click="sortTable('source_url')"
+          @keypress.enter="sortTable('source_url')"
```
Comment (obulat) [inline src/components/VSourcesTable.vue:22]:
```suggestion
          @click="sortTable('source_url')"
```

File: src/components/VSourcesTable.vue (hunk 1 above)
Comment (obulat) [reply src/components/VSourcesTable.vue:60]: 🤯

**Comment by panchovm** [comment]
It looks great. I like how the sort works.

I noticed one thing. The header row should have `dark-charcoal-10` as background color instead of `dark-charcoal-06`.

Great work ✨

File: src/components/VSourcesTable.vue [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  }
+  .table th,
+  .table td {
+    @apply border-dark-charcoal-20;
+  }
+  .table a {
+    @apply text-pink hover:underline;
+  }
+  .table th {
+    @apply bg-dark-charcoal-06 border-t cursor-pointer;
```
Comment (obulat) [inline src/components/VSourcesTable.vue:151]:
```suggestion
    @apply bg-dark-charcoal-10 border-t cursor-pointer;
```

**Comment by sarayourfriend** [comment]
I don't know why the hashed names are being used for the main branch, causing the size diff to fail like that :frowning_face:

**Review by zackkrida — APPROVED** [review]
I can't wait to get rid of `lodash.sortBy`. I can also see some opportunities to make this table a more generic `VDataTable` and decouple if from the sources themselves, but that would be a premature abstraction, I think. This looks great!

**Pull request merged**